src/main/java/seedu/address/model/person/Person.java

```diff
@@ -58,20 +57,53 @@ public Address getAddress() {
         return address;
     }
 
+    public TelegramHandle getTelegramHandle() {
+        return telegramHandle;
+    }
+
     /**
-     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * Returns a copy of the person's group tags.
+     */
+    public GroupTagSet getGroupTags() {
+        return groupTags;
+    }
+
+    /**
+     * Returns an immutable group tag set, which throws {@code UnsupportedOperationException}
      * if modification is attempted.
      */
-    public Set<GroupTag> getGroupTags() {
-        return Collections.unmodifiableSet(groupTags);
+    public Set<GroupTag> getImmutableGroupTags() {
+        return groupTags.getImmutableGroups();
     }
 
-    public TelegramHandle getTelegramHandle() {
-        return telegramHandle;
+    /**
+     * Returns a copy of the person's module tags.
+     */
+    public ModuleTagSet getModuleTags() {
+        return moduleTags;
+    }
+
+    /**
+     * Returns an immutable module tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<ModuleTag> getImmutableModuleTags() {
+        return moduleTags.getImmutableModules();
     }
 
-    public Set<ModuleTag> getModuleTags() {
-        return Collections.unmodifiableSet(moduleTags);
+    /**
+     * Returns an immutable module tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<ModuleTag> getImmutableCommonModuleTags() {
+        return Collections.unmodifiableSet(moduleTags.getImmutableCommonModules());
+    }
```

Comment:
@zichen-3974 Possible common modules method you might want to use in the future.